rust_src/src/base64.rs

```diff
@@ -0,0 +1,127 @@
+extern crate libc;
+extern crate base64;
+
+use base64::base64::{decode, encode};
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ptr;
+
+const MIME_LINE_LENGTH: usize = 76;
```

Comment:
This is also defined in strings.rs, we probably don't want to repeat the definition.